Sources/MapboxCoreNavigation/NavigationEventsManager.swift

```diff
@@ -362,6 +362,7 @@ open class NavigationEventsManager {
         }
         
         sendOutstandingFeedbackEvents(forceAll: true)
+        Thread.sleep(forTimeInterval: 3) // block main thread until queued events are sent to the server
```

Comment:
Are you sure that events library doesn't use main thread to do its work?

Reply:
How difficult would it be to test this behaviour?

Reply:
I've tested it, unfortunately, this fix doesn't seem to work 😿
I will close the PR, we will need some persistence solution from the Events library side.